e2e/menu_items_inline/headings.spec.js

```diff
@@ -0,0 +1,160 @@
+import { test, expect } from "@playwright/test";
+import { VISUAL_EDITOR_URL } from "../helpers/constants.js";
+
+test.beforeEach(async ({ page }) => {
+  await page.goto(VISUAL_EDITOR_URL);
+});
+
+test.describe("H2", () => {
+  test("renders H2 menu items", async ({ page }) => {
+    await page.getByText("H2", { exact: true }).click();
+    await expect(page.locator(".menubar")).toBeVisible();
+    const visibleMenuButtons = [
+      "B",
+      "p",
+      "H3",
+      "“”",
+      "<>",
+      "$A",
+      "$CTA",
+      "$C",
+      "$E",
+      "^",
+      "%",
+    ];
+    const disabledMenuButtons = ["H2", "1.", "-"];
+
+    for (const button of visibleMenuButtons)
+      await expect(page.getByText(button, { exact: true })).toBeVisible();
+    for (const button of disabledMenuButtons)
+      await expect(page.getByText(button, { exact: true })).toBeDisabled();
+  });
+
+  test("loads H2 from the index file in the editor", async ({ page }) => {
+    await expect(
+      page.locator("#editor H2").getByText("This is an H2 heading"),
+    ).toBeVisible();
+  });
+
+  test("should render H2 headings in the editor and clear style after new line when clicking on 'H2' and typing", async ({
+    page,
+  }) => {
+    await page.locator("#editor .ProseMirror.govspeak").focus();
+    await page.keyboard.type("New line\n");
+
+    await page.getByText("New line").click();
+    await page.getByText("H2", { exact: true }).click();
+    await page.getByText("New line").selectText();
+    await page.keyboard.type("Testing H2!\nTesting not H2!\n");
+    await expect(
+      page.locator("#editor h2").getByText("Testing H2!"),
+    ).toBeVisible();
+    await expect(
+      page.locator("#editor h2").getByText("Testing not H2!"),
+    ).not.toBeVisible();
+  });
+
+  test("should toggle H2 headings on for existing paragraph line", async ({
+    page,
+  }) => {
+    await page.locator("#editor .ProseMirror.govspeak").focus();
+    await page.keyboard.type("Testing paragraph\n");
+
+    await page.locator("#editor p").getByText("Testing paragraph").click();
+    await page.getByText("H2", { exact: true }).click();
+    await expect(
+      page.locator("#editor h2").getByText("Testing paragraph"),
+    ).toBeVisible();
+  });
+
+  test("should toggle H2 headings off for existing heading", async ({
```

Comment:
This test feels a bit redundant to me - we're basically testing the "p" command here, I would argue.

Reply:
Agreed to leave this in place as we are treating feature tests as product specifications and ignoring our implementation